rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -1222,6 +1221,35 @@ public JobId copyJobAndResubmitWithGeneralInfo(JobId jobId, Map<String, String>
         return null;
     }
 
+    @Override
+    public Job getJobContent(JobId jobId) throws NotConnectedException, PermissionException, UnknownJobException,
+            JobCreationException, SubmissionClosedException {
+        File tempFile = null;
+        try {
+            final InputStream jobContent = restApi().getJobContent(sid, jobId.value());
+            tempFile = File.createTempFile("jobContent", jobId.value());
+            FileUtils.copyInputStreamToFile(jobContent, tempFile);
+            final Job job = JobFactory.getFactory().createJob(tempFile.getPath());
+            return job;
+        } catch (NotConnectedRestException e) {
+            throw new NotConnectedException(e);
+        } catch (UnknownJobRestException e) {
+            throw new UnknownJobException(e);
+        } catch (PermissionRestException e) {
+            throw new PermissionException(e);
+        } catch (SubmissionClosedRestException e) {
+            throw new SubmissionClosedException(e);
+        } catch (JobCreationRestException e) {
+            throw new JobCreationException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)